ironfish-contracts/contracts/Ironfish.sol

```diff
@@ -4,19 +4,38 @@ pragma solidity ^0.8.18;
 contract Ironfish {
     event Shield(
         bytes32 ironfishAddress,
-        bytes32 ironfishAssetId,
+        uint tokenId,
         address contractAddress,
         uint amount
     );
 
+    event UnShield(
+        address contractAddress,
+        uint tokenId,
+        uint amount
+    );
+
     constructor() {}
 
     function shield(
         bytes32 ironfishAddress,
-        bytes32 ironfishAssetId,
+        uint tokenId,
         uint amount
     ) public {
         // msg.sender is the contract address of the ERC20 token
-        emit Shield(ironfishAddress, ironfishAssetId, msg.sender, amount);
+        emit Shield(ironfishAddress, tokenId, msg.sender, amount);
+    }
+
+    function shield_iron(bytes32 ironfishAddress) public payable {
```

Comment:
nit - since IRON is the native asset I tink that `shield` and `unshield` should be for IRON, and something like `shield_token` would be for custom tokens

Reply:
Hmm I dunno, I think you could argue either way. I think people will generally interact with shield token more, so from a usability perspective might make more sense to have it the way Daniel has it.